### ISSUE TITLE
twisted.conch.test.test_recvline fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8709]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8709|
|Type|enhancement|
|Created|2016-07-29 03:03:30Z|
|Branch|https://github.com/twisted/twisted/tree/8709-rodrigc-conch-recvline-py3|

If I do this:

```
python bin/trial twisted.conch.test.test_recvline
```

I get this:

```
twisted.conch.test.test_recvline
  ArrowsTests
    test_backspace ...                                                     [OK]
    test_delete ...                                                        [OK]
    test_end ...                                                           [OK]
    test_home ...                                                          [OK]
    test_horizontalArrows ...                                              [OK]
    test_insert ...                                                        [OK]
    test_newline ...                                                       [OK]
    test_printableCharacters ...                                           [OK]
    test_typeover ...                                                      [OK]
    test_unprintableCharacters ...                                         [OK]
    test_verticalArrows ...                                                [OK]
  HistoricRecvlineLoopbackSSHTests
    testDownArrow ...                                                   [ERROR]
    testUpArrow ...                                                     [ERROR]
  HistoricRecvlineLoopbackStdioTests
    testDownArrow ...                                                      [OK]
    testUpArrow ...                                                        [OK]
  HistoricRecvlineLoopbackTelnetTests
    testDownArrow ...                                                      [OK]
    testUpArrow ...                                                        [OK]
  RecvlineLoopbackSSHTests
    testBackspace ...                                                   [ERROR]
    testDelete ...                                                      [ERROR]
    testEnd ...                                                         [ERROR]
    testHome ...                                                        [ERROR]
    testInsert ...                                                      [ERROR]
    testLeftArrow ...                                                   [ERROR]
    testRightArrow ...                                                  [ERROR]
    testSimple ...                                                      [ERROR]
    testTypeover ...                                                    [ERROR]
  RecvlineLoopbackStdioTests
    testBackspace ...                                                      [OK]
    testDelete ...                                                         [OK]
    testEnd ...                                                            [OK]
    testHome ...                                                           [OK]
    testInsert ...                                                         [OK]
    testLeftArrow ...                                                      [OK]
    testRightArrow ...                                                     [OK]
    testSimple ...                                                         [OK]
    testTypeover ...                                                       [OK]
  RecvlineLoopbackTelnetTests
    testBackspace ...                                                      [OK]
    testDelete ...                                                         [OK]
    testEnd ...                                                            [OK]
    testHome ...                                                           [OK]
    testInsert ...                                                         [OK]
    testLeftArrow ...                                                      [OK]
    testRightArrow ...                                                     [OK]
    testSimple ...                                                         [OK]
    testTypeover ...                                                       [OK]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/Users/crodrigues/twisted_15/twisted/conch/test/test_recvline.py", line 484, in setUp
    [checkers.InMemoryUsernamePasswordDatabaseDontUse(**{u: p})])
builtins.TypeError: __init__() keywords must be strings

twisted.conch.test.test_recvline.HistoricRecvlineLoopbackSSHTests.testDownArrow
twisted.conch.test.test_recvline.HistoricRecvlineLoopbackSSHTests.testUpArrow
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testBackspace
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testDelete
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testEnd
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testHome
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testInsert
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testLeftArrow
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testRightArrow
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testSimple
twisted.conch.test.test_recvline.RecvlineLoopbackSSHTests.testTypeover
-------------------------------------------------------------------------------
Ran 44 tests in 4.658s

FAILED (errors=11, successes=33)
```

<details><summary>Searchable metadata</summary>

```
trac-id__8709 8709
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8709_rodrigc_conch_recvline_py3 8709-rodrigc-conch-recvline-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469761410462204 1469761410462204
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/444

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_2"></a>|@acabhishek942 commented|
|-|-|

looks good to me. Please merge

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset dbd8cdb2ed148fb87839b360c7b1391a6c189a5f
```
#!CommitTicketReference repository="" revision="dbd8cdb2ed148fb87839b360c7b1391a6c189a5f"
Merge pull request #444 from twisted/8709-rodrigc-conch-recvline-py3

Author: rodrigc
Reviewer:  acabhishek942
Fixes: #8709

Fix twisted.conch.test.test_recvline on Python 3
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
